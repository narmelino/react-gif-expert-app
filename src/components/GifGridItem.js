const GifGridItem = ({title, url}) => (
  <div className="card animate__animated animate__fadeIn">
    <img src={url} alt={title} />
    <p>{title}</p>
  </div>
);

export default GifGridItem;