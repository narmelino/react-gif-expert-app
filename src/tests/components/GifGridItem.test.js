import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas de <GifGritItem />', () => {
  const title = 'One Punch Man Preview GIF';
  const url = 'https://media2.giphy.com/media/arbHBoiUWUgmc/giphy.gif?cid=bce6e586dgm3omnh06gjms04coduyi97cr9db3c968kc7odu&rid=giphy.gif&ct=g';
  const className = 'animate__fadeIn';
  
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  
  test('Debe renderizar el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe tener un pÃ¡rrafo con el title', () => {
    const p = wrapper.find('p');
    expect(p.text()).toBe(title);
  });

  test('Debe tener la imagen igual al url y el alt igual al title', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('Debe tener un div con la clase fadeIn de animate.css', () => {
    const div = wrapper.find('div');
    expect(div.hasClass(className)).toBe(true);
  });
});