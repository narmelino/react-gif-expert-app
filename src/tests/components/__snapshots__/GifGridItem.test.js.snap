// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGritItem Debe renderizar el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="One Punch Man Preview GIF"
    src="https://media2.giphy.com/media/arbHBoiUWUgmc/giphy.gif?cid=bce6e586dgm3omnh06gjms04coduyi97cr9db3c968kc7odu&rid=giphy.gif&ct=g"
  />
  <p>
    One Punch Man Preview GIF
  </p>
</div>
`;
